sycl/doc/extensions/supported/sycl_ext_intel_queue_submission_mode.asciidoc

```diff
@@ -0,0 +1,106 @@
+= sycl_ext_intel_queue_submission_mode
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2023-2023 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 7 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+
+== Status
+
+This extension is implemented and fully supported by {dpcpp}.
+
+
+== Backend support status
+
+This extension is currently implemented in {dpcpp} only for Intel GPU devices
+when using the Level Zero backend.
+
+== Overview
+
+This extension enables specifying one of two submission modes for a queue.
+
+The property `no_immediate_command_list` specifies that submissions to a SYCL
+queue should be collected into groups before actual submission to the hardware.
+
+The property `immediate_command_list` specifies that submissions to a
+SYCL queue should be submitted to the hardware immediately.
+
+When the submission mode is left unspecified a platform-specific default
+is chosen by the implementation.
```

Comment:
This Overview section should be a general introduction to the extension, describing why someone would want to use it.  Don't describe details like the names of the properties here.  Those details should be in the Specification section.  The information in the "Guidelines for using queue submission properties" section could be here in the Overview because it helps a reader understand when to use this extension.  For example:

> When commands are submitted to a SYCL queue that uses the Level Zero backend, those commands can be submitted to the hardware in one of two ways: either through an immediate command list or through a standard command queue.  Commands submitted through an immediate command list are immediately submitted to the device while commands submitted through a standard command queue may be batched with other commands before they are submitted.  By default the implementation chooses a method that works best for most workloads.
>
> In most cases, applications should rely on the default behavior.  However, sometimes it is advantageous for the application to choose one method or the other.  This extension provides a way for applications to select either of these two methods via a queue property.
>
> For example, when kernel runtimes are very short, the submission time on the host may be as long or longer than the actual runtime of the kernel. In this case, doing batched submissions may be preferable so that the submission overhead is amortized over a number of kernel executions.